state/indexer/block/kv/kv.go

```diff
@@ -0,0 +1,490 @@
+package kv
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/google/orderedcode"
+	dbm "github.com/tendermint/tm-db"
+
+	"github.com/celestiaorg/optimint/state/indexer"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/pubsub/query"
```

Comment:
minor note: to future proof your system you may not want to use the tendermint indexer but only the interface then add your own. In tendermint we want to deprecate the current query system and move towards external indexing as this should not be the responsibility of the node

Reply:
The code uses an internal indexer - do you mean that I should also replace the external import of the query package with an internal one?

Reply:
No, saying that this indexer and query language will be deprecated. I know evmos relies on it, but may be good to already think how to better the system instead of absorbing the pitfalls of tendermint.